mobile/src/views/MyICardView.jsx

```diff
@@ -1,34 +1,100 @@
-import React, {useContext} from 'react';
-
+import React, {useContext, useEffect, useState, } from 'react';
+import { View,  RefreshControl, StyleSheet,  ScrollView, ActivityIndicator } from 'react-native';
 import AuthContext from '../context/AuthContext';
 import MyICardPage from '../components/shared/ICardPage';
 import VerificationView from './VerificationView'
 import {NavigationContainer} from '@react-navigation/native';
 import {createNativeStackNavigator} from '@react-navigation/native-stack';
+//import * as StudentFunctions from '../../../backend/src/controllers/Students';
+
 
-let finalStatus = 'inactive';
+let finalStatus;
+let message;
 
 const Stack = createNativeStackNavigator();
 
+import {API_ROUTE, API_KEY} from '@env';
+import { storeUser } from '../utilites/StoreUser';
+
+const authRoute = API_ROUTE;
+const apiKey = API_KEY;
+
 const MyICard = ({navigation}) => {
-  const {user, _} = useContext(AuthContext);
+  const {user, setUser} = useContext(AuthContext);
+  const [checkStatus, setCheckStatus] = useState(true);
+  const [refreshing, setRefreshing] = useState(false);
+
+  const loadUserData = async () => {
+
+    setRefreshing(true);
+    fetch(authRoute + `api/students/${user.id}`, {
+      method: 'GET',
+      headers: {'x-api-key': apiKey, 'jwt-token': user.key},
+    })
+    .then(result => {
+      return result.json();
+    })
+    .then(data => {
+      data["key"] = user.key;
+      data["id"] = data["_id"];
+      data["verification_image"] == ""  || data["verification_image"] == undefined ? data["verification_image"] = "" : null;
```

Comment:
This logic is abit confusing, best to refactor it into two lines for clarity

Reply:
Idk if it's possible to refactor this coz it's vital that I do these particular steps to make sure the user object that is stored everywhere is consistent and that the status checks are done properly. Do you want me to add a comment instead just to clear this up?

Reply:
Just line 40 is the problem, its tough to work out what is happening. If we could do an if statement would be clearer.